### DISCUSSION TITLE
how to store Allure test results into database and generate Allure dashboard by pulling those test results from database

### DISCUSSION THREAD

**Discussion by Ranga-dev** [allure-framework/allure2#2113] [Ideas & Feedback]
**Is your feature request related to a problem? Please describe.**
Seems Allure doesn't have feature to store automation test results into Database.

**Describe the solution you'd like**
If Allure have Database support to store test results, so that it's a very good option to maintain historical test data on database and same it's very useful to show historical trends on Allure dashboard instead of relying on static test results which is generated by Allure command line.

**Describe alternatives you've considered**
We have a requirement where we need to maintain our historical test data on database and generate dashboard by pulling those test results from database. So how to push test results into database by using Allure framework?

**Additional context**
Same need for all the languages (Java, JavaScript, Python, Swift)

**Comment by jvanzyl** [comment]
If someone points me at the interface for storage, I'll try to write a plugin to store historical results of tests.

**Comment by jvanzyl** [comment]
@baev Can you point me at the interface for storing test results?

**Comment by baev** [comment]
There is a product called Allure TestOps that does exactly that. Instead of maintaining all the test data locally, it will upload it to the server in real-time, process and display the reports & advanced analytics. For more details, please follow https://qameta.io/

**Reply by jvanzyl** [reply]
I'd like to implement it myself using either a database or Prometheus where we have a full historical metrics dashboard system. Is there an interface in the open source version that allows creating a custom output/storage implementation?
